Implement server feature checking

Maxmystere opened the pull request:
(_I mostly did this feature for myself but i'm still sharing it if it's relevant for the project_)

Added a test when `Test Connection` button is pressed that will hide item in the menu drawer if the feature is not implemented on Server

![Screenshot_20210524-232912](https://user-images.githubusercontent.com/24446198/119409901-49b58700-bce8-11eb-8ca7-5e87a7f75d42.jpg)

Maxmystere committed:
Implement server feature checking

tzugen commented on ultrasonic/src/main/java/org/moire/ultrasonic/util/Util.java:
```diff
@@ -359,6 +359,20 @@ public static synchronized String formatPercentage(int percent)
+	 * @param value Is true or false
+	 * @return The corresponding mark.
+	 */
+	public static synchronized String boolToMark(boolean value)
```
Doesn't need to be synchronized...

tzugen commented on ultrasonic/src/main/java/org/moire/ultrasonic/util/Util.java:
```diff
@@ -606,6 +620,18 @@ public void onClick(DialogInterface dialog, int i)
 	}
 
+	public static void showDialog(Context context, int icon, int titleId, String message)
+	{
```
Try to use applicationContext() instead of passing around a context

Maxmystere replied:
I might need help on how to do that here

tzugen replied:
I was actually wrong! Unlike toast, AlertDialog actually requires the Activity context!

But I cleaned up the methods a bit (see my commit, hope that was ok)

tzugen commented on ultrasonic/src/main/kotlin/org/moire/ultrasonic/fragment/EditServerFragment.kt:
```diff
@@ -330,17 +338,65 @@ class EditServerFragment : Fragment(), OnBackPressedHandler {
                 pingResponse = subsonicApiClient.api.ping().execute()
                 ApiCallResponseChecker.checkResponseSuccessful(pingResponse)
 
+                updateProgress(String.format(progressString, "⌛", "⌛", "⌛", "⌛"))
```
There is quite a bit of code duplication here. Maybe put the handling of which string to display into the UpdateProgress() function. And does this class ha e a viewmodel? If yes the actual connectio. Checking should go there...

Maxmystere replied:
Updated
It's a bit cleaner now

tzugen reviewed (commented):
Hi I put some annotations in the code. I couldn't test it yet. Is the screen shot showing what you did?

Maxmystere committed:
Fix lintFormat + remove synchronized

Maxmystere committed:
change exception type

tzugen committed:
Cleanup AlertDialog method

The other method was unused, so we can get rid of it.
Also use a lambda for the callback

tzugen committed:
Fix KtLint issue

Maxmystere committed:
Reduce function length

Maxmystere committed:
Merge remote-tracking branch 'base/develop' into check-server-features

Maxmystere committed:
Fix new active server getter

Maxmystere committed:
run ktlint

Maxmystere committed:
Fix length

nitehu committed:
Fixed detekt
Fixed code duplication
Added more info to the Connection Test dialog for the user

nitehu committed:
Fixed ktlint

nitehu committed:
Removed unused string

Maxmystere committed:
Add french translation
